receipt/src/main/java/com/hyperwallet/android/receipt/view/ListReceiptFragment.java

```diff
@@ -0,0 +1,307 @@
+                mTransactionTitle = item.findViewById(R.id.transaction_title);
+                mTransactionTypeIcon = item.findViewById(R.id.transaction_type_icon);
+            }
+
+            void bind(@NonNull final Receipt receipt) {
+                if (CREDIT.equals(receipt.getEntry())) {
+                    mTransactionAmount.setTextColor(mTransactionAmount.getContext()
+                            .getResources().getColor(R.color.positiveColor));
+                    mTransactionAmount.setText(mTransactionAmount.getContext()
+                            .getString(R.string.credit_sign, receipt.getAmount()));
+                    mTransactionTypeIcon.setTextColor(mTransactionTypeIcon.getContext()
+                            .getResources().getColor(R.color.positiveColor));
+                    mTransactionTypeIcon.setBackground(mTransactionTypeIcon.getContext()
+                            .getDrawable(R.drawable.circle_positive));
+                    mTransactionTypeIcon.setText(mTransactionTypeIcon.getContext().getText(R.string.credit));
+                } else if (DEBIT.equals(receipt.getEntry())) {
+                    mTransactionAmount.setTextColor(mTransactionAmount.getContext()
+                            .getResources().getColor(R.color.colorAccent));
+                    mTransactionAmount.setText(mTransactionAmount.getContext()
+                            .getString(R.string.debit_sign, receipt.getAmount()));
+                    mTransactionTypeIcon.setTextColor(mTransactionTypeIcon.getContext()
+                            .getResources().getColor(R.color.colorAccent));
+                    mTransactionTypeIcon.setBackground(mTransactionTypeIcon.getContext()
+                            .getDrawable(R.drawable.circle_negative));
+                    mTransactionTypeIcon.setText(mTransactionTypeIcon.getContext().getText(R.string.debit));
+                }
+
+                mTransactionCurrency.setText(receipt.getCurrency());
+                mTransactionTitle.setText(getTransactionTitle(receipt.getType(),
+                        mTransactionAmount.getText().length(), mTransactionTitle.getContext()));
+                mTransactionDate.setText(DateUtils.toDateFormat(DateUtils.
+                        fromDateTimeString(receipt.getCreatedOn()), CAPTION_DATE_FORMAT));
+            }
+
+            String getTransactionTitle(@NonNull final String receiptType, final int numberOfCharsAlreadyUsed,
+                    @NonNull final Context context) {
+                String showTitle = context.getResources().getString(R.string.unknown_type);
+                int resourceId = context.getResources().getIdentifier(receiptType.toLowerCase(Locale.ROOT), "string",
+                        context.getPackageName());
+                if (resourceId != 0) {
+                    showTitle = context.getResources().getString(resourceId);
+                }
+
+                if (!context.getResources().getBoolean(R.bool.isLandscape)
```

Comment:
@peter-joseph Maybe we can use "ellipsize" ? (https://developer.android.com/reference/android/widget/TextView.html#attr_android:ellipsize) and why MAX_CHARACTERS_FIRST_LINE is 38?